src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,83 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    @BeforeAll
+    static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+    @Test
+    public void register_validUser_ok() {
+        User user = new User("validLogin", "password123", 20);
+        User registeredUser = registrationService.register(user);
+        assertEquals(user, registeredUser);
+    }
+    @Test
+    public void register_existingLogin_notOk() {
+        User user = new User("existingUser", "password123", 20);
+        Storage.people.add(user);
+        User newUser = new User("existingUser", "password123", 20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(newUser);
+        });
+    }
+    @Test
+    public void register_loginTooShort_notOk() {
+        User user = new User("short", "password123", 20);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
```

Comment:
In your tests, it's very important to check all possible edge cases. For example, the minimum password length is 6 characters. In this case, it would be good to test: empty password (0-length - not-ok), 'abc' (3-length - not-ok), 'abcdf' (5-length - not-ok edge case), 'abcdef' (6-length - ok edge case), 'abcdefgh' (8-length - ok). You can have a few not-ok checks in one not-ok test method. Same for ok-test. Testing only 'abc' (3-length - not-ok) and 'abcdefgh' (8-length - ok) is not enough because we do not check how the program behaves with edge cases.